src/Land/Model/soil_heat.jl

```diff
@@ -1,35 +1,251 @@
 ### Soil heat model
+
 export SoilHeatModel, PrescribedTemperatureModel
 
 abstract type AbstractHeatModel <: AbstractSoilComponentModel end
 
 """
-    PrescribedTemperatureModel{FT} <: AbstractHeatModel
+    struct PrescribedTemperatureModel{F1} <: AbstractHeatModel
 
-A model where the temperature is set by the user and not dynamically determined.
+Model structure for a prescribed temperature model.
+
+Document.
 
-This is useful for driving Richard's equation without a back reaction on temperature.
 # Fields
 $(DocStringExtensions.FIELDS)
 """
-Base.@kwdef struct PrescribedTemperatureModel{FT} <: AbstractHeatModel
-    "Prescribed function for temperature"
-    T::FT = FT(0.0)
+struct PrescribedTemperatureModel{F1} <: AbstractHeatModel
+    "Temperature"
+    T::F1
 end
 
 """
-    get_temperature(m::PrescribedTemperatureModel)
+    PrescribedTemperatureModel(
+        T::Function = (aux,t) -> eltype(aux)(0.0)
+    )
+
+Outer constructor for the PrescribedTemperatureModel defining default values, and
+making it so changes to those defaults are supplied via keyword args.
+
+The functions supplied by the user are point-wise evaluated and are 
+evaluated in the Balance Law functions (kernels?) compute_gradient_argument,
+ nodal_update, etc. whenever the prescribed temperature content variables are 
+needed by the water model.
 """
-get_temperature(m::PrescribedTemperatureModel) = m.T
+function PrescribedTemperatureModel(T::Function = (aux, t) -> eltype(aux)(0.0))
+    return PrescribedTemperatureModel{typeof(T)}(T)
+end
 
+"""
+    SoilHeatModel{FT, FiT, BCD, BCN} <: AbstractHeatModel
 
+The necessary components for the Heat Equation in a soil water matrix.
 
+# Fields
+$(DocStringExtensions.FIELDS)
 """
-    SoilHeatModel <: AbstractHeatModel
+struct SoilHeatModel{FT, FiT, BCD, BCN} <: AbstractHeatModel
+    "Initial conditions for temperature"
+    initialT::FiT
+    "Dirichlet BC structure"
+    dirichlet_bc::BCD
+    "Neumann BC structure"
+    neumann_bc::BCN
+end
 
-The balance law for internal energy in soil.
+"""
+    SoilHeatModel(
+        ::Type{FT};
+        initialT::FT = FT(NaN),
+        dirichlet_bc::AbstractBoundaryFunctions = nothing,
+        neumann_bc::AbstractBoundaryFunctions = nothing
+    ) where {FT}
+
+Constructor for the SoilHeatModel. Defaults imply ....
+"""
+function SoilHeatModel(
+    ::Type{FT};
+    initialT = (aux) -> FT(NaN),
+    dirichlet_bc::AbstractBoundaryFunctions = nothing,
+    neumann_bc::AbstractBoundaryFunctions = nothing,
+) where {FT}
+    args = (initialT, dirichlet_bc, neumann_bc)
+    return SoilHeatModel{FT, typeof.(args)...}(args...)
+end
+
+
+#Need these get_temperature functions for SoilHeatModel.
```

Comment:
we need to define get_temperature and get_initial_temperature for the SoilHeatModel in order to run SoilHeat and SoilWater together. it's ok not to include here, I think, but we need them for next time 👍

Reply:
true, and ok! :)